test/framework/multicluster.go

```diff
@@ -23,6 +23,12 @@ func NewMulticlusterE2ETest(t *testing.T, managementCluster *ClusterE2ETest, wor
 	for i, c := range workloadClusters {
 		c.clusterFillers = append(c.clusterFillers, api.WithManagementCluster(managementCluster.ClusterName))
 		c.ClusterName = fmt.Sprintf("%s-w-%d", managementCluster.ClusterName, i)
+
+		if c.GitOpsConfig != nil {
+			c.GitOpsConfig.SetName(c.ClusterName)
+			c.clusterFillers = append(c.clusterFillers, api.WithGitOpsRef(c.ClusterName))
```

Comment:
Could this leave in the `ClusterE2ETest`? (constructor or any of the other "setup" methods)

It seems to be something not tied to the concept of workload clusters and it only needs the `ClusterE2ETest` itself

Reply:
Not sure if i can. The workload cluster name is generated after the flux setup. I need to update GitOpsConfig name after that.

Reply:
avoided overriding gitopsconfig name by always generating new names